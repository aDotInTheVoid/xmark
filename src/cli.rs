use clap::Clap;
use eyre::Result;
use std::path::PathBuf;

#[derive(Debug, Default, Clone)]
pub struct Args {
    pub dir: PathBuf,
    pub create: bool,
    pub templates: Option<PathBuf>,
}

impl Args {
    pub fn parse() -> Result<Self> {
        let ArgsInner {
            mut dir,
            create,
            mut templates,
        } = ArgsInner::parse();
        dir = dir.canonicalize()?;
        templates = match templates {
            Some(p) => Some(p.canonicalize()?),
            None => None,
        };
        Ok(Args {
            dir,
            create,
            templates,
        })
    }
}

#[derive(Clap, Debug)]
struct ArgsInner {
    #[clap(short, long, default_value = ".")]
    pub dir: PathBuf,
    #[clap(long)]
    pub create: bool,
    #[clap(short, long)]
    pub templates: Option<PathBuf>,
}
