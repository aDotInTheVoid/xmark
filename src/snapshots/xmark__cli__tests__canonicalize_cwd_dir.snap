---
source: src/cli.rs
expression: args
---
dir: BASEDIR
create: true
templates: ~
